src/scheduler/work_counter.rs

```diff
@@ -0,0 +1,135 @@
+//! Counter for work packets
+//!
+//! Provides an abstraction and implementations of counters for collecting
+//! work-packet level statistics
+//!
+//! See [`crate::util::statistics`] for collecting statistics over a GC cycle
+use std::time::SystemTime;
+
+/// Common struct for different work counters
+///
+/// Stores the total, min and max of counter readings
+#[derive(Copy, Clone, Debug)]
+pub(super) struct WorkCounterBase {
+    pub(super) total: f64,
+    pub(super) min: f64,
+    pub(super) max: f64,
+}
+
+/// Make [`WorkCounter`] trait objects cloneable
+pub(super) trait WorkCounterClone {
```

Comment:
Do we need this? Can we just implement `Clone` for WorkCounter, or require `WorkerCounter: Clone`?

Reply:
https://doc.rust-lang.org/book/ch17-02-trait-objects.html#object-safety-is-required-for-trait-objects

The `Clone` trait is not object safe.